src/cforth/decomp2.fth

```diff
@@ -122,10 +123,9 @@ variable break-type  variable break-addr   variable where-break
    -1 break-addr !   ( prime stack)
    end-breaks @  breaks  ?do
       i  2@ over   break-addr @ u<  if
-         break-type !  break-addr !  i where-break !
-      else
-         2drop
+         2dup break-type !  break-addr !  i where-break !
       then
+      2drop
```

Comment:
This change seems hardly worth the effort.  Your formulation compiles to slightly shorter code, but the improvement is so slight...